packages/animations/browser/src/dsl/animation_timeline_builder.ts

```diff
@@ -130,10 +130,22 @@ export class AnimationTimelineBuilderVisitor implements AstVisitor {
 
     // this checks to see if an actual animation happened
     const timelines = context.timelines.filter(timeline => timeline.containsAnimation());
-    if (timelines.length && Object.keys(finalStyles).length) {
-      const tl = timelines[timelines.length - 1];
-      if (!tl.allowOnlyTimelineStyles()) {
-        tl.setStyles([finalStyles], null, context.errors, options);
+
+    // note: we just want to apply the final styles for the rootElement, so we do not
+    //       just apply the styles to the last timeline but the last timeline which
+    //       element is the root one (basically *s are replaced with the actual state
+    //       style values only for the root element)
+    let i = timelines.length;
+    let lastRootTimeline;
+    while (--i >= 0 && !lastRootTimeline) {
```

Comment:
I'd personally be in favor of using a for-loop here

```suggestion
    for (let i = timelines.length - 1; !lastRootTimeline && i >= 0; i--) {
```

or perhaps better yet is to introduce a `break` statement after assigning to `lastRootTimeline` below such that this becomes 

```suggestion
    for (let i = timelines.length - 1; i >= 0; i--) {
```

which is the much more familiair reverse-loop, compared to the `while`-loop that it is now.